script.js

```diff
@@ -9,137 +9,343 @@ document.addEventListener("DOMContentLoaded", () => {
-          text-transform: uppercase;
+          width: 1200px;
+          background: gray;
+          padding: 20px;
+          margin: 0 auto;
+          text-align: center;
+          border: 2px solid black;
+          `,
+      });
+      const caption = this.createEl({
+        el: "caption",
+        text: this.tableCaption,
+        styles: `
+          color: gray;
+          font-size: 30px;
           padding: 10px 0px;
           `,
       });
-      trHead.append(th);
-    });
 
-    for (let item of fetchedData) {
-      const tr = createEl({ el: "tr" });
-      const td = createEl({ el: "td", styles: `padding: 10px;` });
-      const input = createCheckbox({
-        el: "input",
-        type: "checkbox",
-        attribute: "id",
-        attributeVal: item.id,
-      });
-      td.append(input);
-      tr.append(td);
-      thNames.slice(1).forEach((thName) => {
-        const td = createEl({
-          el: "td",
-          text: item[thName],
+      const thead = this.createEl({ el: "thead" });
+      const trHead = this.createEl({ el: "tr" });
+      const tbody = this.createEl({ el: "tbody" });
+
+      table.append(caption);
+      table.append(thead);
+      table.append(tbody);
+      thead.append(trHead);
+
+      this.thNames.forEach((title) => {
+        const th = this.createEl({
+          el: "th",
+          text: title,
           styles: `
-          padding: 10px;
-          `,
+            font-size: 18px;
+            text-transform: uppercase;
+            padding: 10px 0px;
+            `,
+        });
+        trHead.append(th);
+      });
+
+      this.fetchedData.forEach((item) => {
+        const tr = this.createEl({ el: "tr" });
+        const td = this.createEl({ el: "td", styles: `padding: 10px;` });
+        const input = this.createCheckbox({
+          el: "input",
+          type: "checkbox",
+          attribute: "id",
+          attributeVal: item.id,
         });
+        td.append(input);
         tr.append(td);
+        this.thNames.slice(1).forEach((thName) => {
+          const td = this.createEl({
+            el: "td",
+            text: item[thName],
+            styles: `
+                  padding: 10px;
+                  `,
+          });
+          tr.append(td);
+        });
+        tbody.append(tr);
       });
-      tbody.append(tr);
-    }
-    document.body.prepend(table);
-  };
+      return document.body.append(table);
+    };
+  }
 
-  const createBtn = () => {
-    const button = createEl({
-      el: "button",
-      text: "delete",
-      styles: `width: 60px; height: 30px; margin: 20px auto; display: block; background: gray; border: none; color: white;`,
+  getData("https://jsonplaceholder.typicode.com/users")
+    .then((data) => {
+      const table = new TableWithDeleting({
+        tableCaption: "Table #1 (Users data)",
+        id: "table1",
+        thNames: ["check", "name", "phone", "website"],
+        fetchedData: data,
+      });
+      return table;
```

Comment:
It isn't a Promise. You don't need to invoke the following methods in a separate "then"